ros_ws/src/autonomous/q_learning/scripts/plotter.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+
+from PyQt4 import QtCore, QtGui
+from parameters import *
+import pyqtgraph as pg
+import rospy
+import sys
+from q_learning.msg import EpisodeResult
+
+
+def update_result_lists(msg):
+    reward_list.append(msg.reward)
+    length_list.append(msg.length)
+
+
+def update_plot():
+    reward_plot.setData(range(len(reward_list))[-50:], reward_list[-50:])
+    length_plot.setData(range(len(length_list))[-50:], length_list[-50:])
```

Comment:
Have you considered removing the timer and updating the plot in `EpisodeResult` callback?

Reply:
I would create a constant for the number of episodes and also increase it to 200 or 400.

Reply:
> Have you considered removing the timer and updating the plot in the `EpisodeResult` callback?

I swear I've tried something like this before and it didn't work because the graph cannot be updated by a different thread. Works now, though.